pytorch_lightning/trainer/supporters.py

```diff
@@ -76,3 +76,17 @@ def _agg_memory(self, how: str):
                 return getattr(self.memory, how)()
             else:
                 return getattr(self.memory[:self.current_idx], how)()
+
+
+class Accumulator(object):
```

Comment:
maybe we should add a different interface here, that just calls these functions. I thought about something like `accumulator += val` for adding another value and `mean = accumulator()` for calculating the accumulated value